### PR TITLE
feat: improve error logging and troubleshooting

### PR DESCRIPTION
**Pull request by Cafe137**

### CODE REVIEW

**Review by nugaon — APPROVED** [review]
noice

**Comment by Cafe137** [comment]
image

**Comment by Cafe137** [comment]
image

File: src/utils/error.ts [hunk 1]
```diff
@@ -1,9 +1,63 @@
...
+
+function isGenericErrorPattern(errorName: string, message: string | unknown): boolean {
+  if (!message || typeof message !== 'string') {
+    return true
+  }
+
+  errorName = errorName.toLowerCase()
+  message = message.toLowerCase()
+
+  return message === errorName || message === `${errorName}: ${errorName}`
```
Comment (nugaon) [inline src/utils/error.ts:62]: For me it is a bit vague what this function checks.

first, at the condition check if `message` is falsy or not string, then it is true.

and the last return I totally lost the tract why the `message` should correspond that format. can you write comments on these or a JSDoc please?

File: src/utils/error.ts [hunk 2]
```diff
@@ -1,9 +1,63 @@
...
+    printer.printError('The command failed with error message: ' + error)
+  } else {
+    printer.printError('The command failed, but there is no error message available.')
+  }
+
+  if ((typeof error === 'string' && error) || message) {
+    printer.printError('')
+    printer.printError('There may be additional information in the Bee logs.')
+  } else {
+    printer.printError('')
```
Comment (nugaon) [inline src/utils/error.ts:49]: why don't you use `\n` instead?

File: src/utils/error.ts [hunk 3]
```diff
@@ -1,9 +1,63 @@
...
+function isNotFoundError(error: unknown): boolean {
+  return hasStatusCode(error, 404)
+}
+
+function isInternalServerError(error: unknown): boolean {
+  return hasStatusCode(error, 500)
+}
+
+export function handleError(error: unknown, options?: BeeErrorOptions): void {
+  const message = getFieldOrNull(error, 'message')
```
Comment (nugaon) [inline src/utils/error.ts:21]: can't it be like this? as I checked you mainly using either this or that to print error message.
```suggestion
  const message = typeof error === 'string' ? error : getFieldOrNull(error, 'message')
```

**Pull request merged**